### PR TITLE
Mat not color

### PR DESCRIPTION
**Pull request by Carlbaum**

### CODE REVIEW

File: include/camera.h [hunk 1]
```diff
@@ -18,16 +18,15 @@
   TODO: We might want to declare this array on the heap
   (no restrictions on size except when RAM is filled up)
 */
-#define WIDTH 1000
-#define HEIGHT 1000
+#define WIDTH 250
+#define HEIGHT 250
```
Comment (novalain) [inline include/camera.h:22]: Why?

File: src/geometry/triangle.cc [hunk 2]
```diff
@@ -35,9 +40,12 @@ bool Triangle::RayIntersection(Ray& ray, float& z) {
   float u = (glm::dot(P, T) / glm::dot(P, E1));
   float v = (glm::dot(Q, D) / glm::dot(P, E1));
 
+  //TODO: I think that t>1 is only valid for the ray that goes directly from the camera
+  //This should probably be changed to t>0 for other rays that has bounced
   if(u >= 0 && v >= 0 && u+v <= 1 && t > 1 && t < z) { //if collision with a triangle closer to cam than before
     Vertex intersection_vertex = (1-u-v)*v0_ + u*v1_ + v*v2_;
-    ray.set_intersection_point(new IntersectionPoint(intersection_vertex, this->normal_, Material(0.f, 0.f, 0.f, this->color_)));
+    //TODO: update the Material to something like: this->material or object->get_material()
```
Comment (novalain) [inline src/geometry/triangle.cc]: I don't get this comment. Whats wrong with just material_ ?

File: include/scene.h [hunk 3]
```diff
@@ -21,6 +21,10 @@ class Scene {
   const std::vector<std::unique_ptr<SceneObject>>& get_objects() const {
     return scene_objects_;
   }
+
+  const std::vector<std::unique_ptr<SceneObject>>& get_lights() const {
```
Comment (novalain) [inline include/scene.h]: `typedef std::vector<std::unique_ptr<SceneObject>> LightList` ?

File: include/triangle.h [hunk 4]
```diff
@@ -3,23 +3,25 @@
...
   Triangle() {}
   Triangle(Vertex v0, Vertex v1, Vertex v2);
-  Triangle(Vertex v0, Vertex v1, Vertex v2, glm::vec3 color);
+  Triangle(Vertex v0, Vertex v1, Vertex v2, ColorDbl color);
+  Triangle(Vertex v0, Vertex v1, Vertex v2, Material material);
 
-  glm::vec3 get_normal() { return normal_; }
-  glm::vec3 get_color() { return color_; }
+  Direction get_normal() { return normal_; }
+  ColorDbl get_color() { return material_.get_color(); }
```
Comment (novalain) [inline include/triangle.h]: Hmmmmm okay, maybe. But I'd prefer using `triangleInstance.get_material().get_color()`

File: include/ray.h [hunk 5]
```diff
@@ -22,7 +21,7 @@ class Ray
   //TODO: implement GetTriangle() ?
   Vertex start() { return v_start_; }
   Vertex end() { return v_end_; }
-  ColorDbl get_color() { return color_;}
+  ColorDbl get_color() { return color_; }
```
Comment (novalain) [inline include/ray.h]: Should not be needed anymore

File: include/light.h [hunk 6]
```diff
@@ -0,0 +1,18 @@
...
+
+#include "scene_object.h"
+
+class Light : public SceneObject {
+private:
+protected:
+  float intensity_;
+  ColorDbl color_;
+public:
+  Light() = default;
```
Comment (novalain) [inline include/light.h]: Why is default constructor needed?

File: include/light.h [hunk 7]
```diff
@@ -0,0 +1,18 @@
...
+  ~Light() {}
```
Comment (novalain) [inline include/light.h]: Doesn't make any sense to have empty destructor

**Review by novalain — CHANGES_REQUESTED** [review]

File: src/point_light.cc [hunk 8]
```diff
@@ -6,15 +6,20 @@ PointLight::PointLight() {
...
   color_ = COLOR_WHITE;
 }
 
-PointLight::PointLight(Vertex position, float intensity) :
-    position_(position), intensity_(intensity), color_(COLOR_WHITE) {
-  // color_ = COLOR_WHITE;
+PointLight::PointLight(Vertex position, float intensity) {
+    position_ = position;
+    intensity_ = intensity;
+    color_ = COLOR_WHITE;
```
Comment (novalain) [inline src/point_light.cc]: Indentation

File: include/camera.h (hunk 1 above)
Comment (Carlbaum) [reply include/camera.h:22]: just for testing.. faster ;)

File: src/geometry/triangle.cc (hunk 2 above)
Comment (Carlbaum) [reply src/geometry/triangle.cc]: oops.. its a leftover from before I actually changed the line under it

File: include/light.h (hunk 6 above)
Comment (Carlbaum) [reply include/light.h]: Cause you said so ^^,

File: src/point_light.cc (hunk 8 above)
Comment (Carlbaum) [reply src/point_light.cc]: arggghhhhh!

Comment (Carlbaum) [reply src/point_light.cc]: ATOM sucks!

File: include/scene.h (hunk 3 above)
Comment (Carlbaum) [reply include/scene.h]: naaaah.. I actually think SceneObject should be replaced with Light in that line..

File: include/camera.h (hunk 1 above)
Comment (novalain) [reply include/camera.h:22]: Smart ;)

File: include/triangle.h (hunk 4 above)
Comment (Carlbaum) [reply include/triangle.h]: Yeah, maybe.. I just thought we could have both options.

File: include/scene.h (hunk 3 above)
Comment (novalain) [reply include/scene.h]: Ok but then Light class needs a virtual destructor as SceneObject has.

File: include/light.h (hunk 6 above)
Comment (novalain) [reply include/light.h]: Nah it doesnt make any sense. Remove :)

**Review by novalain — APPROVED** [review]

**Pull request closed without merging**